addon/mixins/route.js

```diff
@@ -283,17 +278,61 @@ export default Ember.Mixin.create({
 
    @method updateInfinityModel
    @param {Ember.Enumerable} newObjects The new objects to add to the model
-   @return {Ember.Array} returns the updated infinity model
+   @return {DS.RecordArray} returns the updated infinity model
    */
   updateInfinityModel(newObjects) {
```

Comment:
Fair enough. I made this change because the goal is to call `updateInfinityModel` not just after page 2+ but also on page 1 (see https://github.com/hhff/ember-infinity/issues/90). I do think at a minimum we'll want to ensure `infinityModelUpdated` gets called.

On that topic, `updateInfinityModel` is a weird concept to me. What is it? Is it a hook? Is it a call to super? Particularly since this is a mixin, ideally the API would only use hooks that don't depend on a call to super, so users don't have to worry about return value or messing up functionality. The problem with the `infinityModelUpdated` hook however is that it's called outside of the current run loop, so that bound params may not be updated from new results before the next page load is triggered. We could leave `updateInfinityModel` and `infinityModelUpdated` as is, but we'd still need a hook that lets you update the bound params within the same run loop.